moveit_core/CMakeLists.txt

```
@@ -250,9 +259,9 @@ pybind_add_module(pymoveit_core
     planning_scene/src/pyplanning_scene.cpp
     kinematic_constraints/src/pykinematic_constraint.cpp
 )
-target_include_directories(pymoveit_core SYSTEM PRIVATE ${catkin_INCLUDE_DIRS})
-target_link_libraries(pymoveit_core PRIVATE ${pymoveit_libs} ${catkin_LIBRARIES})
-
+target_include_directories(pymoveit_core SYSTEM PRIVATE ${PYTHON_INCLUDE_DIRS} ${pybind11_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS})
+target_link_libraries(pymoveit_core PRIVATE ${pymoveit_libs} ${PYTHON_LIBRARIES} ${catkin_LIBRARIES})
```

```suggestion
target_link_libraries(pymoveit_core PRIVATE ${pymoveit_libs} pybind11::module ${catkin_LIBRARIES})
```
Would it work to use modern cmake targets here to reduce the diff? There are multiple exported targets to choose from

https://github.com/pybind/pybind11/blob/master/tools/pybind11Config.cmake.in#L37